tests/bsim/bluetooth/host/central/src/main.c

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "bstests.h"
+#include "babblekit/testcase.h"
+#include <zephyr/bluetooth/conn.h>
+
+static K_SEM_DEFINE(sem_connected, 0, 1);
+
+static void connected_cb(struct bt_conn *conn, uint8_t err)
+{
+	TEST_ASSERT(conn);
+	TEST_ASSERT(err == BT_HCI_ERR_UNKNOWN_CONN_ID, "Expected connection timeout");
+
+	k_sem_give(&sem_connected);
+	bt_conn_unref(conn);
+}
+
+static struct bt_conn_cb conn_cb = {
+	.connected = connected_cb,
+};
+
+static void test_central_connect_timeout_with_timeout(uint16_t timeout_param,
+						      uint32_t expected_conn_timeout_ms)
+{
+	int err;
+	struct bt_conn *conn;
+
+	bt_addr_le_t peer = {.a.val = {0x01}};
+	const struct bt_conn_le_create_param create_param = {
+		.options = BT_CONN_LE_OPT_NONE,
+		.interval = BT_GAP_SCAN_FAST_INTERVAL,
+		.window = BT_GAP_SCAN_FAST_WINDOW,
+		.interval_coded = 0,
+		.window_coded = 0,
+		.timeout = timeout_param,
```

Comment:
We could consider making `timeout_param` the same unit as `expected_conn_timeout_ms` so this would be
```suggestion
static void test_central_connect_timeout_with_timeout(uint16_t timeout_param_ms,
						      uint32_t expected_conn_timeout_ms)
{
	int err;
	struct bt_conn *conn;

	bt_addr_le_t peer = {.a.val = {0x01}};
	const struct bt_conn_le_create_param create_param = {
		.options = BT_CONN_LE_OPT_NONE,
		.interval = BT_GAP_SCAN_FAST_INTERVAL,
		.window = BT_GAP_SCAN_FAST_WINDOW,
		.interval_coded = 0,
		.window_coded = 0,
		.timeout = timeout_param_ms / 10,
```

In fact, we could probably just have a single value
```suggestion
static void test_central_connect_timeout_with_timeout(uint16_t timeout_param_ms)
{
	int err;
	struct bt_conn *conn;

	bt_addr_le_t peer = {.a.val = {0x01}};
	const struct bt_conn_le_create_param create_param = {
		.options = BT_CONN_LE_OPT_NONE,
		.interval = BT_GAP_SCAN_FAST_INTERVAL,
		.window = BT_GAP_SCAN_FAST_WINDOW,
		.interval_coded = 0,
		.window_coded = 0,
		.timeout = timeout_param_ms / 10,
```

and validate on that, right?